gnome-initial-setup/gis-driver.c

```diff
@@ -911,31 +942,38 @@ window_realize_cb (GtkWidget *widget, gpointer user_data)
 }
 
 static void
-emit_startup_sound (GApplication * app, gpointer user_data)
+startup_sound_cb (EOSHackSoundClient *client, GAsyncResult *res,
+    GisDriver *driver)
 {
-  g_autoptr(GDBusProxy) proxy = NULL;
+  gchar *uuid;
   g_autoptr(GError) error = NULL;
-  const GDBusProxyFlags proxy_flags =
-    G_DBUS_PROXY_FLAGS_DO_NOT_LOAD_PROPERTIES |
-    G_DBUS_PROXY_FLAGS_DO_NOT_CONNECT_SIGNALS;
-
-  proxy = g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SESSION,
-                                         proxy_flags,
-                                         NULL,
-                                         "com.endlessm.HackSoundServer",
-                                         "/com/endlessm/HackSoundServer",
-                                         "com.endlessm.HackSoundServer",
-                                         NULL, &error);
+  GisDriverPrivate *priv = gis_driver_get_instance_private (driver);
+
+  uuid = eos_hack_sound_client_play_finish (priv->sound_client, res, &error);
```

Comment:
Please assign directly to `priv->startup_sound_uuid` since this function returns `NULL` on error.